packages/components/src/SideDrawer/SideDrawer.tsx

```diff
@@ -70,6 +80,11 @@ export function SideDrawer({
         toolbarPortal: ref?.querySelector(toolbar.selector),
         backPortal: ref?.querySelector(backButton.selector),
         footerPortal: ref?.querySelector(footer.selector),
+        components,
+        registerComponent: key =>
```

Comment:
I'm approving to unstick, but I will admit this feels like a whole lot of code/system in order to set the internal component state to 'there is a back button present'. However I can see you've set this up to be extended easily when adding new components to the pattern, and the design of that system is on a solid foundation. 

Also, I'm not huge on patterns being setup when there's only one usable example so far (backButton). In your defense though, I imagine you do this sort of premature optimization because if you don't, you run the risk of showing up to the code in 6 weeks to find 3 new props after `backButtonPresent` for  things like `headerPresent` or `toobarPresentSoPleaseLoadMyFeatureSpecificChange` instead of a single line of `registerComponent('MyNewComponent')`. So long way of saying, it's a lot, but I understand why it's a lot.

Reply:
yeah, it's getting there, i'm hoping this is the last of it with this compound pattern